controller/space.go

```diff
@@ -132,6 +152,24 @@ func (c *SpaceController) Delete(ctx *app.DeleteSpaceContext) error {
 	if err != nil {
 		return jsonapi.JSONErrorResponse(ctx, err)
 	}
+
+	// Try to delete AuthZ resource for the space as part of soft migration from deprecated Keycloak AuthZ API to new OSIO AuthZ API
+	// Old spaces doesn't have any registered resources, so, we don't return an error if unable to find the corresponding resource
+	svc := c.app.ResourceService()
+	err = svc.Delete(ctx, ctx.SpaceID.String())
+	if err != nil {
+		if notFound, _ := errors.IsNotFoundError(err); notFound {
+			log.Warn(ctx, map[string]interface{}{
+				"space_id": ctx.SpaceID,
+			}, "unable to delete authZ space resource: resource not found; that's OK for old spaces")
+		} else {
+			log.Error(ctx, map[string]interface{}{
+				"space_id": ctx.SpaceID,
+			}, "unable to delete authZ space resource")
+			return jsonapi.JSONErrorResponse(ctx, err)
```

Comment:
Usually during soft migration, we don't return an error from the controller if the 'new' code fails. But I think it's fine to do here now since the 'new code' doesn't talk to an external service.